### PR TITLE
#263 Chat scroll animation as optional

### PR DESCRIPTION
**Pull request by NFSS10**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-robin-revamp/pull/267#issuecomment-866939952 |
| Dependencies | -- |
| Decisions | Add `animateScrollToEnd` props that dictates if the chat scroll is/isn't animated |

**Animated:**

https://user-images.githubusercontent.com/22588915/123624942-1d20ed80-d807-11eb-87fb-2b20f2c0f7bb.mov

**Not animated:**

https://user-images.githubusercontent.com/22588915/123624938-1c885700-d807-11eb-83e0-656d7211a77d.mov

### CODE REVIEW

File: react/components/organisms/chat/chat.js [hunk 1]
```diff
@@ -59,7 +61,7 @@ export class Chat extends PureComponent {
     };
 
     scrollToEnd = () => {
-        this.scrollViewComponent.scrollToEnd({ animated: true });
+        this.scrollViewComponent.scrollToEnd({ animated: this.props.scrollAnimated });
```
Comment (gcandal) [inline react/components/organisms/chat/chat.js:64]: Who is calling `scrollToEnd`?

Comment (NFSS10) [reply react/components/organisms/chat/chat.js:64]: React `ScrollView` component: https://github.com/ripe-tech/ripe-components-react-native/blob/a10530729b7bb7463f36af2877e5cf8f3e96203a/react/components/organisms/chat/chat.js#L127
and inside the `_onNewMessage`method: https://github.com/ripe-tech/ripe-components-react-native/blob/a10530729b7bb7463f36af2877e5cf8f3e96203a/react/components/organisms/chat/chat.js#L79

Comment (gcandal) [reply react/components/organisms/chat/chat.js:64]: Does this need to be a prop then? Isn't it enough to be a `scrollToEnd` argument that you manipulate in `chat.js`?

Comment (NFSS10) [reply react/components/organisms/chat/chat.js:64]: I added this prop so I can disable the scroll animation when the chat messages are being loaded. It's related to the enhancement you asked me to do (https://github.com/ripe-tech/ripe-robin-revamp/pull/267#issuecomment-866939952)

Comment (gcandal) [reply react/components/organisms/chat/chat.js:64]: Ok rename to `animateScrollToEnd` then, and it's good to go

**Review by gcandal — APPROVED** [review]

**Review by joamag — APPROVED** [review]

**Pull request merged**